tests/cephfs/cephfs_upgrade/cephfs_post_upgrade_validation.py

```diff
@@ -741,11 +758,245 @@ def mds_config_test(mds_req_params):
+            snap_list = qs_info[sv]["snap_list"]
+            cmd = f"mkdir {mnt_pt}/new_dir_{rand_str};cp -r {mnt_pt}/.snap/*{snap_name}*/* {mnt_pt}/new_dir_{rand_str}"
+            for client_tmp in cg_quiesce_params["clients"]:
+                if client == client_tmp.node.hostname:
+                    out, rc = client_tmp.exec_command(sudo=True, cmd=cmd)
+                    log.info(out)
+                    cmd = (
+                        f"diff {mnt_pt}/.snap/*{snap_name}* {mnt_pt}/new_dir_{rand_str}"
+                    )
+                    out, rc = client_tmp.exec_command(sudo=True, cmd=cmd)
+                    log.info(out)
+                    log.info(
+                        f"Restore from old snapshot {snap_name} to AFS suceeds after CG quiesce test on {sv}"
+                    )
+
+    log.info(f"cg_test_io_status : {cg_test_io_status.value}")
+
+    if p.is_alive():
+        proc_stop = 0
+        log.info("CG IO is running after quiesce lifecycle")
+        io_run_time *= 2
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=io_run_time)
+        while (datetime.datetime.now() < end_time) and (proc_stop == 0):
+            if p.is_alive():
+                time.sleep(10)
+            else:
+                proc_stop = 1
+        if proc_stop == 1:
+            log.info("CG IO completed")
+        elif proc_stop == 0:
+            raise Exception("CG IO has NOT completed")
+    else:
+        log.info(
+            f"WARN:CG IO test completed early during quiesce lifecycle with await on qs_set {qs_id_val}"
+        )
+
+    mnt_pt_list = []
+    log.info(f"Perform cleanup for {qs_set}")
+    for qs_member in qs_member_dict:
+        mnt_pt_list.append(qs_member_dict[qs_member]["mount_point"])
+    log.info("Remove CG IO files and unmount")
+    cg_snap_util.cleanup_cg_io(cg_io_client, mnt_pt_list)
+    mnt_pt_list.clear()
+
+    snap_name = f"cg_snap_{rand_str}"
+    log.info("Remove CG snapshots")
+    for qs_member in qs_member_dict:
+        snap_list = snap_qs_dict[qs_member]
+        if qs_member_dict[qs_member].get("group_name"):
+            group_name = qs_member_dict[qs_member]["group_name"]
+            for snap_name in snap_list:
+                fs_util.remove_snapshot(
+                    cg_client,
+                    vol_name,
+                    qs_member,
+                    snap_name,
+                    validate=True,
+                    group_name=group_name,
+                )
+        else:
+            for snap_name in snap_list:
+                fs_util.remove_snapshot(
+                    cg_client, vol_name, qs_member, snap_name, validate=True
+                )
+
+    if cg_test_io_status.value == 1:
+        log.error(
+            f"CG IO test exits with failure during quiesce lifecycle with await on qs_set-{qs_id_val}"
+        )
+        test_fail = 1
+
+    if test_fail == 1:
+        log.error("FAIL: CG quiesce lifecycle on set of existing and new subvolumes")
```

Comment:
Can we differentiate if the quiesce failed on existing pre-upgrade ones or new subvolumes post upgrade?

Reply:
Yes Neha,
i am reporting quiesce test failure based on IO validation tool, and in this tool, i am reporting subvolume name if IO validation fails, so we will know which subvolume caused failure, as subvolume name has pre_upgrade in its name for existing volumes